python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_shouldnt_accept_noncallable_object(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf)
```

Comment:
Different positive scenarios are quite well covered by the [existing tests](https://github.com/apache/spark/blob/master/python/pyspark/sql/tests.py#L271-L310).

`self.assertRaises` actually calls its argument when it is executed. But we should really test `UserDefinedFunction` directly here, not `udf`. Let me correct that.